crates/consensus-logic/src/csm/client_transition.rs

```diff
@@ -372,196 +416,203 @@ mod tests {
-                &blkids[0..idx + 1]
-            );
-        }
-
-        // genesis + 2
-        {
-            let height = genesis + 2;
-            let idx = (height - horizon) as usize;
-            let l1_block_id = l1_chain[idx].block_hash().into();
-            let event = SyncEvent::L1Block(height, l1_block_id);
-
-            let output = process_event(&state, &event, database.as_ref(), &params).unwrap();
-
-            let expected_writes = [ClientStateWrite::AcceptL1Block(l1_block_id)];
-            let expected_actions = [];
-
-            assert_eq!(output.writes(), expected_writes);
-            assert_eq!(output.actions(), expected_actions);
-
-            operation::apply_writes_to_state(&mut state, output.writes().iter().cloned());
-
-            assert!(!state.is_chain_active());
-            assert_eq!(state.most_recent_l1_block(), Some(&l1_block_id));
-            assert_eq!(state.next_exp_l1_block(), height + 1);
-            assert_eq!(
-                state.l1_view().local_unaccepted_blocks(),
-                &blkids[0..idx + 1]
-            );
-        }
-
-        // genesis + 3, where we should lock in genesis
-        {
-            let height = genesis + 3;
-
-            let idx = (height - horizon) as usize;
-            let genesis_id = l1_chain[(genesis - horizon) as usize].block_hash().into();
-            let l1_block_id = l1_chain[idx as usize].block_hash().into();
-
-            let event1 = SyncEvent::L1BlockGenesis(height, l1_verification_state.clone());
-            let event2 = SyncEvent::L1Block(height, l1_block_id);
-
-            let output1 = process_event(&state, &event1, database.as_ref(), &params).unwrap();
-            let output2 = process_event(&state, &event2, database.as_ref(), &params).unwrap();
-
-            let genesis_block = genesis::make_genesis_block(&params);
-            let genesis_blockid = genesis_block.header().get_blockid();
-
-            let expected_writes1 = [
-                ClientStateWrite::ActivateChain,
-                ClientStateWrite::UpdateVerificationState(l1_verification_state.clone()),
-                ClientStateWrite::ReplaceSync(Box::new(SyncState::from_genesis_blkid(
-                    genesis_blockid,
-                ))),
-            ];
-            let expected_writes2 = [ClientStateWrite::AcceptL1Block(l1_block_id)];
-
-            let expected_actions1 = [SyncAction::L2Genesis(genesis_id)];
-            let expected_actions2 = [];
-
-            assert_eq!(output1.writes(), expected_writes1);
-            assert_eq!(output1.actions(), expected_actions1);
-
-            assert_eq!(output2.writes(), expected_writes2);
-            assert_eq!(output2.actions(), expected_actions2);
-
-            operation::apply_writes_to_state(&mut state, output1.writes().iter().cloned());
-            operation::apply_writes_to_state(&mut state, output2.writes().iter().cloned());
+        let test_cases = vec![
```

Comment:
It seems to me like the test cases are more structured now, but are not very readable. Maybe some parameterized helper functions could make these more readable.